File: solutions/codeforces/190C.py
import sys
sys.setrecursionlimit(1000000)

n = int(raw_input())
A = raw_input().split()
l = len(A)

err = False
t = 0

def parseType():
    global t, err
    if t >= len(A):
        err = True
        return ''
    t += 1
    if A[t-1] == 'pair':
        return 'pair<' + parseType() + ',' + parseType() + '>'
    else:
        return 'int'

ans = parseType()
print ans if err == False and t == l else "Error occurred"

File: solutions/codeforces/190C.cpp
#include <stdio.h>
#include <string.h>

int N;
int A[100010];
bool err = false;
int t = 0;
int l = 0;
char ab[1000000];
int p = 0;;

void parseType()
{
        if (t >= l) {
                err = true;
                return;
        }
        t++;
        if (A[t-1] == 'p') {
                strcpy(ab+p, "pair<"); p += 5;
                parseType();
                strcpy(ab+p, ","); p += 1;
                parseType();
                strcpy(ab+p, ">"); p += 1;
        }
        else {
                strcpy(ab+p, "int"); p += 3;
        }
}

int main()
{
        scanf("%d", &N);
        while (true) {
                char buf[10];
                if (scanf("%s", buf) != 1)
                        break;
                A[l] = buf[0];
                l++;
        }
        parseType();
        printf("%s\n", (!err && t == l ? ab : "Error occurred"));

        return 0;
}
